vm/src/stdlib/os.rs

```diff
@@ -1542,6 +1542,49 @@ mod posix {
             .map_err(|err| err.into_pyexception(vm))
     }
 
+    #[pyfunction]
+    fn execve(
+        path: PyPathLike,
+        args: Either<PyListRef, PyTupleRef>,
+        env: PyDictRef,
+        vm: &VirtualMachine,
+    ) -> PyResult<()> {
+        let path = ffi::CString::new(path.into_bytes())
+            .map_err(|_| vm.new_value_error("embedded null character".to_owned()))?;
+
+        let args: Vec<ffi::CString> = match args {
+            Either::A(list) => vm.extract_elements(list.as_object())?,
+            Either::B(tuple) => vm.extract_elements(tuple.as_object())?,
+        };
+        let args: Vec<&ffi::CStr> = args.iter().map(|entry| entry.as_c_str()).collect();
+
+        let first = args
+            .first()
+            .ok_or_else(|| vm.new_value_error("execv() arg 2 must not be empty".to_owned()))?;
+
+        if first.to_bytes().is_empty() {
+            return Err(vm.new_value_error("execv() arg2 first element cannot be empty".to_owned()));
+        }
+
+        let env = env
+            .into_iter()
+            .map(|(k, v)| -> PyResult<_> {
+                let (key, value) = (
+                    PyPathLike::try_from_object(&vm, k)?,
+                    PyPathLike::try_from_object(&vm, v)?,
+                );
+                ffi::CString::new(format!("{}={}", key.path.display(), value.path.display()))
+                    .map_err(|_| vm.new_value_error("embedded null character".to_owned()))
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let env: Vec<&ffi::CStr> = env.iter().map(|entry| entry.as_c_str()).collect();
+
+        unistd::execve(&path, &args, &env)
+            .map(|_ok| ())
+            .map_err(|err| err.into_pyexception(vm))
```

Comment:
```suggestion
        unistd::execve(&path, &args, &env)
            .map_err(|err| err.into_pyexception(vm))?
        Ok(())
```

Reply:
what difference does this make?

Reply:
applied changes https://github.com/RustPython/RustPython/pull/2101/commits/f1a7014183786035dd695d4248def2f433eda8ea

Reply:
Just for human readability. Because the ok result is not interesting for human reader at that time. We can reduce one `map` by returning `Ok(())` at the bottom

Reply:
that is quite understandable, thanks